File: SPOJ/candy.cc
#include <iostream>
#include <vector>

using namespace std;

int main(){

	int n;

	while (true){
		cin >> n;

		if (n == -1) break;

		vector<int> v (n);
		int sum = 0;

		for (int i=0; i<n; i++){
			cin >> v[i];
			sum += v[i];
		}

		int t = sum/n;

		if (t*n != sum){
			cout << "-1\n";
			continue;
		}

		int ans = 0;

		for (int i=0; i<n; i++){
			if (v[i] > t){
				ans += v[i]-t;
			}
		}

		cout << ans << endl;


	}

	return 0;
}

File: SPOJ/candy.py
#!/usr/bin/env python 

def main():
	while True:

		n = raw_input().split (' ');
		n = int(n[0])

		if (n == -1):
			break
	
		numbers = []

		for i in range (0, n):
			num = raw_input().split (' ');
			num = int(num[0])
			numbers.append (num)
		
		sum = 0

		for i in range (0, n):
			sum = sum + numbers[i]
	
		
		t = sum/n
		ans = 0

		if (t*n == sum):
			for i in range (0, n):
				if (t > numbers[i]):
					ans = ans + t - numbers[i]
			print ans
		else:
			print "-1"



main()
